File: 2015/Days/01.cs
using _2015.Input;
using System;
using System.Linq;

namespace _2015.Days
{
    class _01
    {
        public void Run()
        {
            var instructions = Helper.GetAsSingleLine("01.txt");
            Console.WriteLine(instructions.Length);
            var floor = instructions.Aggregate(0, (acc, curr) => acc + (curr == '(' ? 1 : -1));
            Console.WriteLine(floor);

            var index = 0;
            var currentFloor = 0;

            foreach (var instruction in instructions)
            {
                index++;
                currentFloor += instruction == '(' ? 1 : -1;
                if (currentFloor == -1)
                {
                    break;
                }
            }

            Console.WriteLine(index);
        }
    }
}

File: 2015/days/01.ts
import { getAsSingleLine } from '../input';
import { success } from '../utils/logger';

export default async () => {
  const instructions = getAsSingleLine('01.txt').split('');

  const finalFloor = instructions.reduce(
    (acc, curr) => acc + calcFloorLevel(curr),
    0
  );

  success(`Part 1: ${finalFloor}`);

  let currentFloor = 0;

  for (let i = 0; i < instructions.length; i++) {
    currentFloor += calcFloorLevel(instructions[i] as string);

    if (currentFloor === -1) {
      success(`Part 2: ${i + 1}`);
      break;
    }
  }
};

const calcFloorLevel = (value: string): number => (value === '(' ? 1 : -1);
